lib/rubocop/cop/style/redundant_begin.rb

```diff
@@ -46,9 +46,11 @@ def autocorrect(node)
 
             corrector.replace(
               range_with_surrounding_space(node.loc.expression),
-              range_with_surrounding_space(
-                child.loc.expression
-              ).source.gsub(/^[ \t]{#{indent_diff}}/, '')
+              range_with_surrounding_space(node.loc.expression)
+                .source
+                .gsub(/\A\s*?begin\s*?(;|$)/, '')
```

Comment:
Too many regexps are a recipe for bugs IMO. It's always preferable to manipulate the nodes directly. @jonas054 Thoughts?